ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/transactions/NewPooledTransactionHashesMessageProcessorTest.java

```diff
@@ -170,17 +189,181 @@ public void shouldNotScheduleGetPooledTransactionsTaskTwice() {
+    assertThat(frontier.getHash())
+        .isEqualTo(
+            Hash.fromHexString(
+                "0x0000000000000000000000000000000000000000000000000000000000000001"));
+    assertThat(frontier.getType()).hasValue(TransactionType.FRONTIER);
+    assertThat(frontier.getSize()).hasValue(1);
+
+    final TransactionAnnouncement accessList = announcementList.get(1);
+    assertThat(accessList.getHash())
+        .isEqualTo(
+            Hash.fromHexString(
+                "0x0000000000000000000000000000000000000000000000000000000000000002"));
+    assertThat(accessList.getType()).hasValue(TransactionType.ACCESS_LIST);
+    assertThat(accessList.getSize()).hasValue(2);
+
+    final TransactionAnnouncement eip1559 = announcementList.get(2);
+    assertThat(eip1559.getHash())
+        .isEqualTo(
+            Hash.fromHexString(
+                "0x0000000000000000000000000000000000000000000000000000000000000003"));
+    assertThat(eip1559.getType()).hasValue(TransactionType.EIP1559);
+    assertThat(eip1559.getSize()).hasValue(3);
+  }
+
+  @Test
+  public void shouldEncodeAndDecodeTransactionAnnouncement_Eth66() {
+    final Transaction t1 = generator.transaction(TransactionType.FRONTIER);
+    final Transaction t2 = generator.transaction(TransactionType.ACCESS_LIST);
+    final Transaction t3 = generator.transaction(TransactionType.EIP1559);
+    final List<Transaction> list = List.of(t1, t2, t3);
+    final Bytes bytes = getEncoder(EthProtocol.ETH66).encode(list);
+
+    final List<TransactionAnnouncement> announcementList =
+        getDecoder(EthProtocol.ETH66).decode(RLP.input(bytes));
+
+    for (int i = 0; i < announcementList.size(); i++) {
+      final TransactionAnnouncement announcement = announcementList.get(i);
+      assertThat(announcement.getHash()).isEqualTo(list.get(i).getHash());
+      assertThat(announcement.getType()).isEmpty();
+      assertThat(announcement.getType()).isEmpty();
+    }
+  }
+
+  @Test
+  public void shouldEncodeAndDecodeTransactionAnnouncement_Eth68() {
+    final Transaction t1 = generator.transaction(TransactionType.FRONTIER);
+    final Transaction t2 = generator.transaction(TransactionType.ACCESS_LIST);
+    final Transaction t3 = generator.transaction(TransactionType.EIP1559);
+
+    final List<Transaction> list = List.of(t1, t2, t3);
+    final Bytes bytes = getEncoder(EthProtocol.ETH68).encode(list);
+
+    final List<TransactionAnnouncement> announcementList =
+        getDecoder(EthProtocol.ETH68).decode(RLP.input(bytes));
+
+    assertThat(announcementList.size()).isEqualTo(list.size());
+
+    for (final Transaction transaction : list) {
+      final TransactionAnnouncement announcement = announcementList.get(list.indexOf(transaction));
+      assertThat(announcement.getHash()).isEqualTo(transaction.getHash());
+      assertThat(announcement.getType()).hasValue(transaction.getType());
+      assertThat(announcement.getSize()).hasValue(transaction.calculateSize());
+    }
+  }
```

Comment:
I suggest to add tests that stress the parsing with malformed messages, like arrays not of the same type, missing elements and wrong types, to make sure them are intecepted and handled

Reply:
will do